include/envoy/network/dns.h

```diff
@@ -39,7 +39,8 @@ class DnsResolver {
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name, uint32_t port,
```

Comment:
@0x1997 before implementing, can you propose the interfaces and configuration method (how the config will look). We can agree on that and then you can implement. That will be more efficient.